packages/blocks/src/root-block/edgeless/edgeless-root-block.ts

```diff
@@ -200,6 +203,23 @@ export class EdgelessRootBlockComponent extends BlockComponent<
       .catch(console.error);
   }
 
+  private _initLayerUpdateEffect() {
+    const updateLayers = requestThrottledConnectedFrame(() => {
+      const blocks =
+        this.renderRoot?.querySelectorAll<GfxBlockComponent>(
```

Comment:
So instead of using `requestUpdate`, here we switch to querying all GFX blocks by DOM and update their index on every layer update. Are there some potential DOM querying performance concerns?

If it's just a temp workaround, it's still acceptable for fixing fatal issues. But in this case, maybe we need to mark it with comments.

Reply:
A better approach is to implement this logic within the `GfxBlockComponent` so that elements can update themselves. But we can't do this because some block does not extends the `GfxBlockComponent`. 

This is a temp workaround. I'll put a label on this.